### DISCUSSION TITLE
SubSonic Server Support

### DISCUSSION THREAD

**Discussion by R1D3R175** [MissingCore/Music#41] [Ideas]
Would really love to connect to my SubSonic server :)

**Comment by Leetram519** [comment]
Kind of necroposting here, but I'd be happy to help develop this, if this is an eventuality !

**Reply by cyanChill** [reply]
There's some hurdles for this:
1. Does the media get detected by [`Expo MediaLibrary`](https://docs.expo.dev/versions/latest/sdk/media-library/)?
2. Will you always be connected to the server (so the files are always detected)?
   - This is mainly due to us cleaning up the database of any tracks that we don't find on each app launch.
3. Can the URI be played by [`react-native-track-player`](https://github.com/doublesymmetry/react-native-track-player) (which can play media from a URL)?

Point 1 can probably be checked whenever I update how we mimic "folder" structure (and thus relax the filter of what tracks we'll "allow" to be checked) as we currently take a shortcut of limiting to the internal or sd card storage paths such as `/storage/emulated/0` & `/storage/0A08-1F1A` (`/storage/0A08-1F1A` should be the sd card path, with the characters after `/storage/` being different between devices).

**Comment by Bhuvisj** [comment]
What is meant by subsonic

**Reply by Leetram519** [reply]
Subsonic is a music server API that's used by things like Navidrome. It's for self-hosted music repositories. https://www.subsonic.org/pages/index.jsp

**Reply by Bhuvisj** [reply]
Thank you Broskie
